packages/langium/test/validation/grammar-validation.test.ts

```diff
@@ -69,7 +63,7 @@ describe('Validate that cross-references are not named as "name"', () => {
     });
 
     test('Named crossReference warning', () => {
-        expectWarning(validationData, 'We recommend not to use the "name" property for cross-references.');
+        expectWarning(validationData, 'The "name" property is not recommended for cross-references.');
```

Comment:
I do not like to compare with the error message text. Think about rewording the messages. You would have to touch also the test. Error codes are an alternative. But I also see the overhead in maintaining them.
But another argument FOR error codes could be localization/translation into other languages.

Reply:
What I like about using the actual diagnostic message in the test is that the test becomes well readable. I understand the concern about changing the message text, but I think adapting test expectations is ok in that case.